File: 1-two-sum/two-sum.cpp

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++)
        {
            int more = target-nums[i];  
            
            if(mp.find(more)!=mp.end())
            {
                return {i,mp[more]};
            }
            mp[nums[i]]=i;
        }
        return {-1,-1}; 
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s={}
        for i in range(len(nums)):
            a=target-nums[i]
            if a in s:
                return (s[a],i)
            s[nums[i]]=i